frontend/express/public/javascripts/countly/countly.views.js

```diff
@@ -638,410 +638,186 @@ window.UserView = countlyView.extend({
-            var sDaysPercentage = countlyCommon.formatNumber((100 * sDays) / totals[2], 2);
-            sDaysPercentage = isNaN(sDaysPercentage) ? 0 : sDaysPercentage;
-
-            var allMultiplier = 0.8;
-            var tDaysMultiplier = 0.8;
-            var sDaysMultiplier = 0.8;
-
-            if (allPercentage > 80) {
-                allMultiplier = 0.65;
-            }
-
-            if (tDaysPercentage > 80) {
-                tDaysMultiplier = 0.65;
-            }
-
-            if (sDaysPercentage > 80) {
-                sDaysMultiplier = 0.65;
-            }
-
-            chartData.push({
-                l: ticks[iTick][1],
-                a_count: all,
-                td_count: tDays,
-                sd_count: sDays,
-                a: "<div style='float:left;min-width: 40px'>" + countlyCommon.formatNumber(all) + "</div><div class='percent-bar' style='width:" + (allPercentage * allMultiplier) + "%'></div>" + allPercentage + "%",
-                td: "<div style='float:left;min-width: 40px'>" + countlyCommon.formatNumber(tDays) + "</div><div class='percent-bar' style='width:" + (tDaysPercentage * tDaysMultiplier) + "%'></div>" + tDaysPercentage + "%",
-                sd: "<div style='float:left;min-width: 40px'>" + countlyCommon.formatNumber(sDays) + "</div><div class='percent-bar' style='width:" + (sDaysPercentage * sDaysMultiplier) + "%'></div>" + sDaysPercentage + "%"
+    },
+    computed: {
+        userActivity: function() {
+            return this.$store.state.countlyUserActivity.userActivity;
+        },
+        seriesTotal: function() {
+            return this.$store.state.countlyUserActivity.seriesTotal;
+        },
+
+        userActivityOptions: function() {
+            return {
+                legend: {
+                    top: "bottom",
+                    padding: [0, 0, 20, 0]
+                },
+                xAxis: {
+                    type: "category",
+                    data: this.xAxisUserActivitySessionBuckets
+                },
+                yAxis: {
+                    type: "value",
+                },
+                series: this.yAxisUserActivityCountSeries
+            };
+        },
+        xAxisUserActivitySessionBuckets: function() {
+            return this.$store.state.countlyUserActivity.nonEmptyBuckets;
+        },
+        yAxisUserActivityCountSeries: function() {
+            var self = this;
+            return Object.keys(this.userActivity).map(function(userActivityKey) {
+                return {
+                    data: self.userActivity[userActivityKey].map(function(item) {
+                        return item.count;
+                    }),
+                    type: "bar",
+                    name: self.barChartItemsLegends[userActivityKey],
+                    itemStyle: {
+                        borderRadius: [4, 4, 0, 0],
+                        color: self.barChartItemsColors[userActivityKey]
+                    },
```

Comment:
bar charts auto selects a color from an array of provided colors.
i dont think you need to provide color aswell.
also i think i should add borderRadius as default to the bar chart.